.github/workflows/release.yml

```
@@ -0,0 +1,158 @@
+            echo "Invalid release version format. Expected format: x.y.z or x.y.z-qualifier"
+            exit 1
+          fi
+          if [[ "${{ github.event.inputs.nextVersion }}" =~ ^[0-9]+\.[0-9]+\.[0-9]+(-[a-zA-Z0-9]+)?$ && ! "${{ github.event.inputs.nextVersion }}" =~ -SNAPSHOT$ ]]; then
+            echo NEXT_VERSION="${{ github.event.inputs.nextVersion }}-SNAPSHOT" >> $GITHUB_ENV
+            echo "Next development version: $NEXT_VERSION"
+          else
+            echo "Invalid next development version format. Expected format: x.y.z or x.y.z-qualifier"
+            exit 1
+          fi
+          echo TAG_NAME="authmgr-${RELEASE_VERSION}" >> $GITHUB_ENV
+
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          fetch-depth: 0 # Fetch all history for all branches and tags
+          ref: main # Checkout the main branch
+
+      - name: Verify current version
+        run: |
+          if [[ ! -f version.txt ]]; then
+            echo "version.txt file not found!"
+            exit 1
+          fi
+          current_version=$(cat version.txt)
+          if [[ ! "$current_version" =~ -SNAPSHOT$ ]]; then
+              echo "Current version is not a SNAPSHOT version: $current_version"
+              exit 1
+          fi
+
+      - name: Configure Git
+        run: |
+          git config --global user.email "authmgr-release-workflow-noreply@dremio.com"
+          git config --global user.name "AuthManager Release Workflow [bot]"
+
+      - name: Update version for release
+        run: |
+          echo "${RELEASE_VERSION}" > version.txt
+          git add version.txt
+          git commit -a -m "chore(release): release version ${RELEASE_VERSION}"
+          git tag -a -m "chore(release): tag version ${RELEASE_VERSION}" ${TAG_NAME}
+
+      - name: Push tag
+        run: |
+          if [[ "${{ github.event.inputs.dryRun }}" == "false" ]]; then
+            echo "Pushing tag ${TAG_NAME}."
+            git push origin ${TAG_NAME}
+          else
+            echo "Dry run enabled, not pushing tag."
+          fi
+
+      - name: Set up JDK 21
+        uses: actions/setup-java@v4
+        with:
+          distribution: temurin
+          java-version: 21
+
+      - name: Build with Gradle
+        run: ./gradlew build publish -x intTest
+
+      - name: JReleaser full release
+        env:
+          JRELEASER_DRY_RUN: ${{ github.event.inputs.dryRun }}
+          JRELEASER_TAG_NAME: ${TAG_NAME}
+          JRELEASER_MAVENCENTRAL_STAGE: FULL
+          JRELEASER_GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          JRELEASER_MAVENCENTRAL_USERNAME: ${{ secrets.DEVBOT_CENTRAL_USERNAME }}
+          JRELEASER_MAVENCENTRAL_PASSWORD: ${{ secrets.DEVBOT_CENTRAL_PASSWORD }}
+          JRELEASER_GPG_PASSPHRASE: ${{ secrets.DEVBOT_GPG_PASSPHRASE }}
+          JRELEASER_GPG_SECRET_KEY: ${{ secrets.DEVBOT_GPG_PRIVATE_KEY }}
+        run: ./gradlew jreleaserFullRelease
```

WDYT of 
```suggestion
        run: ./gradlew jreleaserFullRelease --info
```
until we've seen the release work smoothly. I'm not sure about JReleaser, but sometimes you run into weird failures with meaningless error messages - with `--info` we would have at least some contextual information (hope so at least).
It's sad, but there's no CI or "integration test" for these things - chicken-egg-problem.